src/printer.js

```diff
@@ -3410,10 +3410,19 @@ function isFirstStatement(path) {
   return body && body[0] === node;
 }
 
+function removeEmptyStatementsOnTail(stmts) {
+  let index = stmts.length - 1;
+  while (index >= 0 && stmts[index].type === "EmptyStatement") {
+    index--;
+  }
+
+  return stmts.slice(0, index + 1);
+}
+
 function isLastStatement(path) {
   const parent = path.getParentNode();
   const node = path.getValue();
-  const body = parent.body;
+  const body = removeEmptyStatementsOnTail(parent.body);
```

Comment:
at the first time, I thought about this.
- `filter` is more readable and maintainable.
- my current implement is a bit faster, don't have loop whole array.

it's trade-off. Let me update